Clustering features with phenotype selector

mauromiguelm opened the pull request:
this fixes #261  

- coupled to changes already merged into `playbase master`.
- add selector to cluster heatmap
- fix bugs that appeared when number of selected features were equal to 1 or 0.

It would be good practice to give feedback and communicate user that opened this issue.

ivokwee commented:
If you put the selector on the right settings, then all plots should adhere to these settings. Also the phenoplots and feature ranking. Otherwise the phenotype selector should be in the heatmap options only. Please try with the problem-datasets of Sooheon "sooheon-26mar-failed-contrast" it has 30+ phenotypes.

mauromiguelm commented:
**additions:**

- `featurerank` plot now responsive to Phenotypes input
- `phenoplot` plot now responsive to Phenotypes input
- removed code that overwrittes `selected_phenotypes` user input
- fixed several bugs that crashed plots, as plots were not prepared to have single (or reduced) number of phenotypes
